### PR TITLE
Try: bundle WP_REST_Global_Styles_Controller class instead of inheriting per WordPress version

### PR DESCRIPTION
**Pull request by ramonjd**
## What?
Similar to:

- https://github.com/WordPress/gutenberg/pull/46579

This PR bundles rest global styles controller code into one class `WP_REST_Global_Styles_Controller` to avoid conflicts, and to make sure the response always reflects changes from `WP_Theme_JSON_Gutenberg`.

The `WP_REST_Global_Styles_Controller` is the endpoint that fetches the contents of the theme global styles custom post from the database.

## Why?
Context for change: https://github.com/WordPress/gutenberg/pull/53451/files#r1289969201

For many of the reasons cited in https://github.com/WordPress/gutenberg/pull/46579, but also because the underlying response is directly affected by changes to `WP_Theme_JSON_Gutenberg`.

## How?
Removing all extensions in `lib/compat` and creating an "evergreen" file under `/lib`.

## Testing Instructions
Check that all tests pass on CI.

Ensure both classic (TwentyTwenty) and block themes (TwentyTwentyThree) load properly and that any presets are loaded and work correctly. 

For block themes, perform some common global styles-related functions, e.g., 

In the site editor, check that the endpoint returns the current merged theme.json. Run this command in the browser console: 

```js
await wp.apiFetch( { url: `/index.php?rest_route=%2Fwp%2Fv2%2Fglobal-styles%2F${ wp.data.select('core').__experimentalGetCurrentGlobalStylesId() }&_locale=user` } );
```

In the site editor, make some changes to the global styles (e.g.,: select a different style variation, add a custom color palette). Refresh the page and verify they are applied everywhere (editor, front-end).

Check that the revisions count in the global styles revisions drop down is accurate

### CODE REVIEW

**Comment by ramonjd** [comment]
@michalczaplinski Sorry, I think I missed something in relation to 'behaviors'. The E2E tests are failing. 

Trying to find it, but am code blind. 😅

**Comment by ramonjd** [comment]
> @michalczaplinski Sorry, I think I missed something in relation to 'behaviors'. The E2E tests are failing.

Never mind. User error. Forgot to import a file and got too excited 🤦🏻 All is well.

File: lib/class-wp-rest-global-styles-controller-gutenberg.php [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+ *
+ * Changes to this class should be synced to the corresponding class
+ * in WordPress core: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php.
+ *
+ * @package    gutenberg
+ * @subpackage REST_API
+ * @since 5.9.0
+ */
+
+if ( class_exists( 'WP_REST_Global_Styles_Controller_Gutenberg' ) ) {
```
Comment (oandregal) [inline lib/class-wp-rest-global-styles-controller-gutenberg.php:13]: Why would we need to check for this? Is there a different place where we expect this class to be declared other than this file?

Comment (ramonjd) [reply lib/class-wp-rest-global-styles-controller-gutenberg.php:13]: No reason. I was just copying what we do in [WP_Theme_Json_Gutenberg](https://github.com/WordPress/gutenberg/blob/111c7e05f6a2f9e9dddf6f8b8b534fe35caaaa5f/lib/class-wp-theme-json-gutenberg.php#L9-L9).

I assumed it was just protocol or something 😄

Comment (oandregal) [reply lib/class-wp-rest-global-styles-controller-gutenberg.php:13]: I don't know why we do it there, either! 🙈

Comment (oandregal) [reply lib/class-wp-rest-global-styles-controller-gutenberg.php:13]: It actually may be "protocol" now? https://github.com/WordPress/gutenberg/pull/52696

Comment (ramonjd) [reply lib/class-wp-rest-global-styles-controller-gutenberg.php:13]: > It actually may be "protocol" now? https://github.com/WordPress/gutenberg/pull/52696

That'd be it 😄

**Review by oandregal — APPROVED** [review]

**Pull request merged**